sdk/core/azure-core/src/test/java/com/azure/core/util/BinaryDataTest.java

```diff
@@ -287,6 +292,34 @@ public void createFromFluxLazy() {
         );
     }
 
+    @Test
+    public void wrapFluxLazy() {
+        final byte[] data = "Doe".getBytes(StandardCharsets.UTF_8);
+        final Flux<ByteBuffer> dataFlux = Flux.defer(() -> Flux.just(ByteBuffer.wrap(data), ByteBuffer.wrap(data)));
+        final byte[] expected = "DoeDoe".getBytes(StandardCharsets.UTF_8);
+
+        // Act & Assert
+        Arrays.asList((long) expected.length, null).forEach(
```

Comment:
I'd split this into two separate tests to make it clearer that length is based on what is passed by default. And possibly change the order a bit to:

- Create BinaryData
- Validate initial length is equal to provided length
- Validate toBytes returns the expected byte array
- Validate BinaryData length is equal to expected length

Reply:
I copied several other tests from the other flux API but it looks as if it doesn't quite make sense here, as it's checking some things that don't apply to this API. I'll revisit the test in general.